test/project_types/extension/features/argo_config_test.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+require 'test_helper'
+require 'project_types/extension/extension_test_helpers'
+
+module Extension
+  module Features
+    class ArgoConfigTest < MiniTest::Test
+      def setup
+        super
+        File.stubs(:size?).returns(true)
+        ShopifyCli::ProjectType.load_type(:extension)
+      end
+
+      def test_parses_and_symbolizes_yaml_hash
+        value = {}
+        another_value = 10
+        yml = { "value": value, "another_value": another_value }
+        YAML.stubs(:load_file).returns(yml)
+
+        parsed_config = ArgoConfig.parse_yaml(@context, [:value, :another_value])
+
+        assert_equal(value, parsed_config[:value])
+        assert_equal(another_value, parsed_config[:another_value])
+      end
+
+      def test_aborts_when_yaml_is_invalid
+        Psych::SyntaxError.any_instance.stubs(:initialize)
```

Comment:
@robin-drexler this is generating some warnings in the test output:

```
/Users/andyw8/.gem/ruby/2.5.1/gems/mocha-1.7.0/lib/mocha/mock.rb:351: warning: undefining `initialize' may cause serious problems
/Users/andyw8/.gem/ruby/2.5.1/gems/mocha-1.7.0/lib/mocha/any_instance_method.rb:44: warning: removing `initialize' may cause serious problems
```

Reply:
thanks for letting me know! I have created an issue on our board, but am not sure yet when we'll be able to get to it.

Reply:
I've fixed it in #919